modules/claims_api/app/controllers/claims_api/v0/forms/power_of_attorney_controller.rb

```diff
@@ -125,6 +142,52 @@ def validation_success
             }
           }
         end
+
+        def current_poa_begin_date
+          return nil if current_poa.try(:begin_date).blank?
+
+          Date.strptime(current_poa.begin_date, '%m/%d/%Y')
+        end
+
+        def current_poa
+          @current_poa ||= BGS::PowerOfAttorneyVerifier.new(target_veteran).current_poa
+        end
+
+        def current_poa_code
+          current_poa.try(:code)
+        end
+
+        def previous_poa_code
+          @previous_poa_code ||= BGS::PowerOfAttorneyVerifier.new(target_veteran).previous_poa_code
+        end
+
+        def build_representative_info(poa_code)
+          if poa_code_in_organization?(poa_code)
+            veteran_service_organization = ::Veteran::Service::Organization.find_by(poa: poa_code)
+            raise 'Veteran Service Organization not found' if veteran_service_organization.blank?
+
+            {
+              name: veteran_service_organization.name,
+              phone_number: veteran_service_organization.phone
+            }
+          else
+            representative = ::Veteran::Service::Representative.where('? = ANY(poa_codes)', poa_code).first
+            raise 'Power of Attorney not found' if representative.blank?
+
+            {
+              name: "#{representative.first_name} #{representative.last_name}",
+              phone_number: representative.phone
+            }
+          end
+        end
+
+        def poa_code_in_organization?(poa_code)
+          representative = ::Veteran::Service::Representative.where('? = ANY(poa_codes)', poa_code).first
+          raise 'Power of Attorney not found' if representative.blank?
+          return false if representative.user_types.blank?
+
+          representative.user_types.include?('veteran_service_officer')
+        end
```

Comment:
bit of a bummer to have this logic in 3 separate places. One thing to have it separate between v0 and v1 if its wildly different, but then there is the third place where it was before this pr.